src/msg/async/ProtocolV2.cc

```diff
@@ -1422,7 +1422,7 @@ CtPtr ProtocolV2::handle_message() {
 
   // we need to get the size before std::moving segments data
   const size_t cur_msg_size = get_current_msg_size();
-  auto msg_frame = MessageFrame::Decode(std::move(rx_segments_data));
+  auto msg_frame = MessageFrame::Decode(rx_segments_data);
```

Comment:
It stood out because `Decode` takes a `static_vector`, which behaves like `std::array`, not `std::vector`.  We move the bufferlists individually and `rx_segments_data` still has the same number of elements on return.  Taking by l-lvalue reference seemed a bit clearer to me.